src/asmcnc/production/z_head_mechanics_jig/z_head_mechanics.py

```diff
@@ -0,0 +1,369 @@
+                Clock.schedule_once(self.record_up_values, 0.4)
+            else:
+                if self.m.s.sg_z_motor_axis != -999:
+                    self.sg_values_down.append(self.m.s.sg_z_motor_axis)
+                    self.z_pos_values_down.append(self.m.mpos_z())
+                Clock.schedule_once(self.record_down_values, 0.1)
+
+    def record_up_values(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.phase_two()
+            else:
+                if self.m.s.sg_z_motor_axis != -999:
+                    self.sg_values_up.append(self.m.s.sg_z_motor_axis)
+                    self.z_pos_values_up.append(self.m.mpos_z())
+                Clock.schedule_once(self.record_up_values, 0.1)
+
+
+    def phase_two(self):
+        if self.test_running:
+            self.m.set_motor_current("Z", 13)
+            self.m.jog_absolute_single_axis('Z', self.z_axis_max_travel, self.z_axis_max_speed)
+            Clock.schedule_once(self.continue_phase_two, 0.4)
+
+    def continue_phase_two(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.jog_absolute_single_axis('Z', -1, self.z_axis_max_speed)
+                Clock.schedule_once(self.finish_test, 0.4)
+            else:
+                Clock.schedule_once(self.continue_phase_two, 0.1)
+
+    def finish_test(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.set_motor_current("Z", 25)
+                self.display_results()
+                self.reset_after_stop()
+            else:
+                Clock.schedule_once(self.finish_test, 0.1)
+
+    def display_results(self):
+        self.load_up_peak.text = str(max(self.sg_values_up))
+        self.load_down_peak.text = str(max(self.sg_values_down))
+        self.load_up_average.text = str(sum(self.sg_values_up) / len(self.sg_values_up))
+        self.load_down_average.text = str(sum(self.sg_values_down) / len(self.sg_values_down))
+
+        plt.rcParams["figure.figsize"] = (7.9,3.55)
+        plt.plot(self.z_pos_values_down, self.sg_values_down, 'b', label='Z SG Down')
+        plt.plot(self.z_pos_values_up, self.sg_values_up, 'r', label='Z SG Up')
+        plt.legend()
+        plt.title('Z motor load vs Z coordinate')
+        plt.xlabel('Z coordinate, mm')
+        plt.ylabel('Z load')
+        ax = plt.gca()
+        ax.set_ylim([0, 200])
+        ax.set_xlim([min(self.z_pos_values_down + self.z_pos_values_up), max(self.z_pos_values_down + self.z_pos_values_up)])
+        plt.tight_layout()
+        plt.grid()
+        plt.savefig('./asmcnc/production/z_head_mechanics_jig/z_head_mechanics_jig_graph.png')
+        plt.close()
+        self.load_graph.source = './asmcnc/production/z_head_mechanics_jig/z_head_mechanics_jig_graph.png'
+        self.load_graph.reload()
+        self.load_graph.opacity = 1
+
+
+    def stop(self):
```

Comment:
This won't prevent any scheduled events from continuing to run (i.e. anything that's on a Clock). Possible that movements stop, GRBL gets reset, and then events will carry on & start new movements.